src/config.rs

```diff
@@ -8,92 +8,111 @@ use crate::TileImage;
 pub struct Config {
     pub tilesize: u32, // 256
     pub zoomlevel: u8, // eg 5
+    pub parentzoomlevel: Option<u8>,
+    pub indexforzoom: u8,
     pub preset: Option<u8>,
-    pub startzoomrangetoslice: u8,
-    pub endzoomrangetoslice: u8,
-    pub starttargetrange: u32,
-    pub endtargetrange: u32,
+    pub zoomrangetoslice: RangeInclusive<u8>,
+    pub targetrangetoslice: RangeInclusive<u32>,
 }
 
 impl Config {
     pub fn new(
         // $1
-        tilesize: u32,                            // 256
-        zoomlevel: u8,                            // eg 5
+        tilesize: u32, // 256
+        zoomlevel: u8, // eg 5
+        parentzoomlevel: Option<u8>,
+        indexforzoom: u8,                         // eg 1
         zoomrange: Option<RangeInclusive<u8>>,    // eg 0 - 5
         targetrange: Option<RangeInclusive<u32>>, //eg 0 - 500
         preset: Option<u8>,
     ) -> Self {
-        let zomr = zoomrange.unwrap_or(zoomlevel..=zoomlevel);
-        // total number of tiles required in zoomrange
-        let mut totaltiles = 0;
-        zomr.clone().for_each(|x| {
-            totaltiles += 1 << (x * 2);
-        });
-        // number of tiles sliced
-        let tilessliced = match &targetrange {
-            Some(targetrange) => *targetrange.start(),
-            None => 0,
-        };
-        // number of tiles to slice
-        let tilestoslice = match &targetrange {
-            Some(targetrange) => *targetrange.end() - tilessliced,
-            None => totaltiles,
-        };
-        if tilestoslice > totaltiles {
-            panic!("Target range value cannot be over than the total number of tiles within zoom range.");
-        }
-
-        // zoom level to start
-        let mut startzoomrangetoslice: u8 = *zomr.clone().start();
-        // tile index to start
-        let mut starttargetrange: u32 = 0;
-        // zoom level to stop
-        let mut endzoomrangetoslice: u8 = *zomr.clone().end();
-        // tile index to stop
-        let mut endtargetrange: u32 = 0;
-
-        // total of tiles in previous zoom levels
-        let mut tilessum = 0;
-        // calculte startzoomrangetoslice and starttargetrange
-        for i in zomr.clone() {
-            // number of tiles in this zoom level
-            let currentzoomtiles = 1 << (i * 2);
-            if tilessum + currentzoomtiles > tilessliced {
-                startzoomrangetoslice = i;
-                starttargetrange = tilessliced - tilessum;
-                break;
+        if let Some(parentzoomlevelvalue) = parentzoomlevel {
+            if parentzoomlevelvalue > zoomlevel {
+                Config {
+                    tilesize,
+                    zoomlevel,
+                    parentzoomlevel,
+                    indexforzoom,
```

Comment:
There's no restrictions on `indexforzoom`. This means that if I have `parentzoomlevel` of 2, and `zoomlevel` of 1 (implying 4 subimages with Morton indexes between 0 and 3), I can still put in `--indexforzoom 255` and get some wacky tile numbering out the other end.

We can fairly assume that if the parent zoom level is 2 and the sub image is zoom level 1, then all of the other subimages will be zoom level 1 and there will be 4 of them, and so putting in a `indexforzoom` that's larger than 3 should throw some sort of error.